### PR TITLE
Publish Falco container image to AWS ECR Public registry

### PR DESCRIPTION
**Pull request by leodido**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area build

**What this PR does / why we need it**:

Opening this PR with @jonahjon.

The goal here is to publish the Falco containers images (both the master one and the release one) to the AWS ECR Public registry 📦 

As proposed by @jonahjon during yesterday's community call.

In fact, as discussed, it can be very useful for our community to be able to download Falco container images from different registries, especially since the latest changes to the Docker Hub.

TODOs:

- [x] Publish container image (`falcosecurity/falco:master`) for the `master` branch
- [x] Publish container image (`falcosecurity/falco:<version>`) for the releases (git tags)
- [x] Make these steps depending on those publishing docker images to the Docker Hub
- [x] Obtain `falcosecurity` slug on the AWS ECR Public registry

The following steps could be added in a follow-up PR:

- Publish container image (`falcosecurity:falco-no-driver:master`)
- Publish container image (`falcosecurity:falco-no-driver:<version>`)
- Publish container image (`falcosecurity:falco-driver-loader:master`)
- Publish container image (`falcosecurity:falco-driver-loader:<version>`)

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:

Note this does not change the status of the container images published to Docker Hub, which remains "official support" as per [docs](https://github.com/falcosecurity/falco/blob/master/proposals/20200506-artifacts-scope-part-1.md#images).

**Does this PR introduce a user-facing change?**:

```release-note
build: falcosecurity/falco:master also available on the AWS ECR Public registry
build: falcosecurity/falco:latest also available on the AWS ECR Public registry
```

### CODE REVIEW

**Comment by leodido** [comment]
First available Falco container image here! https://gallery.ecr.aws/b4t6c0y6/falco 🎉

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:470]: Substitute this line with the following one before to merge this in.

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:559]:
```suggestion
            docker push "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}"
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:557]:
```suggestion
            docker tag "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}" public.ecr.aws/falcosecurity/falco:latest
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:559]:
```suggestion
            docker push "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}"
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:558]:
```suggestion
            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:556]:
```suggestion
            docker build --build-arg VERSION_BUCKET=deb --build-arg FALCO_VERSION=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}" docker/falco
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:474]:
```suggestion
            docker push "public.ecr.aws/falcosecurity/falco:master"
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:473]:
```suggestion
            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:472]:
```suggestion
            docker build --build-arg VERSION_BUCKET=deb-dev --build-arg FALCO_VERSION=${FALCO_VERSION} -t "public.ecr.aws/falcosecurity/falco:master" docker/falco
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:471]:
```suggestion
            FALCO_VERSION=$(cat /build/release/userspace/falco/config_falco.h | grep 'FALCO_VERSION ' | cut -d' ' -f3 | sed -e 's/^"//' -e 's/"$//')
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:470]:
```suggestion
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:626]:
```suggestion
              only: master
```

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:628]:
```suggestion
            - publish/docker-dev
```

**Review by leodido — COMMENTED** [review]
Self-review to prepare the changes to submit once the slug `falcosecurity` is ready

File: .circleci/config.yml
Comment (jonahjon) [reply .circleci/config.yml:472]: +1 looks awesome

**Review by jonahjon — APPROVED** [review]

File: .circleci/config.yml
Comment (leodido) [inline .circleci/config.yml:559]:
```suggestion
            docker push "public.ecr.aws/falcosecurity/falco:latest"
```

**Review by fntlnz — APPROVED** [review]

**Review by leogr — COMMENTED** [review]
It seems good to me, but it'd be better to re-use the same docker build to have the same image digest, IMHO.

Is there any reason to push a different build? Am I missing something? :thinking:

**Comment by leogr** [comment]
/hold
to get other pending PRs merged
feel free to un-hold at any time

**Comment by leodido** [comment]
@leogr it seems to me DLC (docker layer caching) is false by default ([ref](https://circleci.com/docs/2.0/docker-layer-caching/#remote-docker-environment)).

Anyway, we can always improve it (and I agree with your point about image SHA) later.

**Comment by leogr** [comment]
> @leogr it seems to me DLC (docker layer caching) is false by default ([ref](https://circleci.com/docs/2.0/docker-layer-caching/#remote-docker-environment)).
> 
> Anyway, we can always improve it (and I agree with your point about image SHA) later.

Ok, got it. We can fix it later.

**Review by leogr — APPROVED** [review]
/hold cancel

**Pull request merged**
